### PR TITLE
Rename 'Track' to 'Theme'

### LINKED ISSUE #1: Change “Track” to “Theme”
Change the default option from “General - No Suggested Track” to be “Other -- describe below” and trigger a short answer field for the proposer to use to fill in their talk’s theme.

### PR DESCRIPTION
**Pull request by joshpuetz**
Renames usages of 'Track' to 'Theme' in the UI.  For example:

CleanShot 2021-06-03 at 10 47 39@2x

Also adds a 'Suggested Theme' field. This input shows only if 'Other' theme is picked during proposal creation:

CleanShot 2021-06-04 at 11 01 16

A helper is used to properly display suggested themes through the application: 

CleanShot 2021-06-04 at 11 05 28@2x

Closes https://github.com/forem/cfp-app/issues/1

### CODE REVIEW

**Review by nickytonline — APPROVED** [review]
Search and replace LGTM! 🚢

**Review by fdocr — APPROVED** [review]
The decision of using the target branch `main-codenewbie` and not `main` for this PR is intentional, right? Just asking to make sure it's not a typo or something 🙂

**Comment by joshpuetz** [comment]
> The decision of using the target branch `main-codenewbie` and not `main` for this PR is intentional, right? Just asking to make sure it's not a typo or something 🙂

Correct! I'm using `main-codenewbie` for changes that we aren't contributing back to `rubycentral/cfp-app`, and keeping `main` for pulling upstream changes!

**Comment by joshpuetz** [comment]
Taking this back to draft for a bit: I forgot the requirement to change the default option!

**Comment by joshpuetz** [comment]
Ready for re-review!

File: app/assets/javascripts/proposal.js [hunk 1]
```diff
@@ -80,4 +80,21 @@ $(function() {
       }
     });
   }
+
+  // Show/hide suggested theme input
+  $('select[id="theme_select"]').change(function() {
+    if($(this).val() === ""){
+      $('div.proposal_suggested_theme').show(200)
```
Comment (fdocr) [inline app/assets/javascripts/proposal.js:87]: JQuery 🥰

**Review by fdocr — APPROVED** [review]
Gave this a quick spin and looks good 🙌

**Pull request merged**